cmd/system-probe/config/config.go

```diff
@@ -176,6 +177,15 @@ func load() (*Config, error) {
 		npmEnabled = true
 	}
 
+	deprecatedEnableHttpMonitoringKey := key(netNS, "enable_http_monitoring")
+	enableHttpMonitoringKey := key(smNS, "enable_http_monitoring")
+
+	if cfg.IsSet(deprecatedEnableHttpMonitoringKey) {
+		log.Infof("%q is deprecated, use %q instead",
+			deprecatedEnableHttpMonitoringKey, enableHttpMonitoringKey)
+		cfg.Set(enableHttpMonitoringKey, cfg.GetBool(deprecatedEnableHttpMonitoringKey))
```

Comment:
I think you need to check if `enableHttpMonitoringKey` is set first, otherwise the deprecated setting will override it (in the case that both are set).

Reply:
Done - `enableHttpMonitoringKey` will now take precedence and the tests have been updated accordingly. Note that the default value had to be removed from `service_monitoring_config.enable_http_monitoring` to ensure that it can be determined whether the user has set it or not.